Fix Group dailies reset not working.

prabhjotsbhatia-ca opened the pull request:
[//]: # (Note: See https://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #2 

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)
- Fixed module export in team-cron.js 
- Created new run-team-cron.js to transpile team-cron.js , also added mongodb connection
- Created habiticateamcron to run cron every hour
- Updated Dockerfile to include changes and automatically run cron.

awinterstein committed:
Grant subscription to every user during registration and make first user an admin.

awinterstein committed:
Disables payment for group plans.

awinterstein committed:
Configurable registration only by invite.

awinterstein committed:
Buy gems with gold instead of money from the header bar.

awinterstein committed:
Disables payment scripts setup.

awinterstein committed:
Disables analytics.

awinterstein committed:
Cleanup of the settings pages.

awinterstein committed:
Removes links from the header menu.

awinterstein committed:
Removes "social" logins from the registration and login pages.

awinterstein committed:
Do not show promo events.

awinterstein committed:
Use https instead of ssh for getting a depenceny from github.

awinterstein committed:
The user should just own every background (no need to buy).

awinterstein committed:
Added Dockerfile and workflow to create and publish production containers.

Co-authored-by: Saik0Shinigami

awinterstein committed:
Do not show the modal dialog for external links.

awinterstein committed:
Fixed issue that logging was not actually disabled,
when no transport were configured in the production environment

awinterstein committed:
Disabled email sending, if no mail server is configured.

awinterstein committed:
Allow to leave out some configurations even in production.

awinterstein committed:
Removed unused Facebook parameters from the configuration example

awinterstein committed:
Send emails via SMTP instead of the Mandrill webservice.

awinterstein committed:
Removes information from subscription settings
that do not make sense without payed subscriptions.

awinterstein committed:
Removes most parts of the footer.

prabhjotsbhatia-ca committed:
fix group dailies not reset #2

prabhjotsbhatia-ca committed:
Clean up

awinterstein commented:
Thank you for fixing this already!

I somehow can't get it work with the container right now. When the cronjob is executed, it cannot connect to the database. Could it be that the environment variables are not available to the script, when it is executed via the cronjob?

prabhjotsbhatia-ca committed:
Accommodate new docker file URIs and remove hardcoded values

prabhjotsbhatia-ca commented:
@awinterstein you're right. The new docker file version removes direct port access to the DB, and I had hardcoded that URI. Fixed in the latest commit. Also removed hardcoded SESSION SECRET values - these are now read directly from the config.json file.

prabhjotsbhatia-ca committed:
Merge branch 'self-host' into prabhjot-patch

awinterstein commented:
Great, it seems to work now!
